gce/burst-training/README.md

```diff
@@ -0,0 +1,242 @@
+
+1. The `python census-analysis.py` call must be made with the right command line
+   arguments.
+
+2. The Compute Engine instance should be shut down immediately after the `python
+   censys-analysis.py` call has returned. This will keep Compute Engine charges
+   to a minimum.
+
+We use two Compute Engine mechanisms to achieve this behavior:
+
+1. [Instance
+   metadata](https://cloud.google.com/compute/docs/storing-retrieving-metadata)
+   -- These are key-value pairs of strings that you can set on a Compute Engine
+   VM. In burst training, we use them to set values for command line arguments
+   as well as to deliver the training code to the instance. The additional
+   benefit to using metadata like this is that they are visible on your
+   instances either through the Cloud Console or through `gcloud compute
+   instances describe <INSTANCE_NAME>`.
+
+2. [Startup scripts](https://cloud.google.com/compute/docs/startupscript) --
+   This is a script that is executed immediately once a Compute Engine instance
+   has started. In burst training, as we are using a Linux image (although you
+   can alter this as necessary), we use a
+   [bash](https://www.gnu.org/software/bash/) script which:
+
+   - Loads the appropriate instance metadata
+
+   - Downloads the trainer code to the VM from Google Cloud Storage
+
+   - Runs the training job specifying the appropriate command line arguments
+     from the metadata
+
+   - Shuts down the instance once the training is complete
+
+The [training script](./train.sh) shows this process in action. Once you have
+built your VM image, you can run this script to execute your first burst
+training job.
+
+
+## Customization
+
+You can adapt the procedure described here to perform burst training of your own
+models. Moreover, you should be able to do so with only slight modifications to
+the code. Here is a checklist:
+
+1. Modify [image.sh](./image.sh) to install whatever software you require.
+
+2. Run [build-image.sh](./build-image.sh) to create a new image in your desired
+   `IMAGE_FAMILY`.
+
+3. Upload your data and your trainer to Google Cloud Storage. Although, in this
+   example, our trainer consisted of a single Python file, you can also use a
+   full Python package. Note that you may have to modify your trainer to accept
+   parameters from the command line.
+
+4. Decide on what instance metadata you will use for your burst training.
+
+5. Modify [census-startup.sh](./census-startup.sh) to
+
+   - Load this metadata
+
+   - Download your trainer script or package
+
+   - Run the trainer with the appropriate parameters
+
+   Do not remove the final shutdown command unless you want the instance to keep
+   running even after training has completed.
+
+6. Modify [train.sh](./train.sh) to
+
+   - Create an instance of the appropriate [machine type](https://cloud.google.com/compute/docs/machine-types).
+     Note that you can also specify a custom machine type.
```

Comment:
maybe include a separate `train-custom-machine-type.sh` that illustrates how to do that.

Reply:
Considered it but want to keep the number of files to a minimum. This is again something that I would like to do only if we get user feedback indicating that not having this file is a significant source of friction.

Reply:
Is this alright with you @dizcology ?